### PR TITLE
add hibernate framework

### PR DESCRIPTION
**Pull request by ymilenin**

### CODE REVIEW

File: src/main/java/dao/impl/hibernate/ProductDAOHibImpl.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            logger.error("Error when try delete product from DB from hibernate", e);
+        }
+    }
+
+    @Override
+    public void updateProduct(Product oldProduct, Product newProduct) {
```
Comment (Sofasmile) [inline src/main/java/dao/impl/hibernate/ProductDAOHibImpl.java]: better create method update with one parameter, without oldProduct

File: src/main/java/dao/impl/hibernate/ProductDAOHibImpl.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            }
+            logger.error("Error when try update product in DB from hibernate", e);
+        }
+    }
+
+    @Override
+    public List<Product> getAllProduct() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("FROM Product", Product.class).list();
+        }
```
Comment (Sofasmile) [inline src/main/java/dao/impl/hibernate/ProductDAOHibImpl.java]: what about catch?

File: src/main/java/dao/impl/hibernate/UserDAOHibImpl.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+            return session.createQuery("FROM User", User.class).list();
+        }
+    }
+
+    @Override
+    public Optional<User> getUserByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            User user = session.byNaturalId(User.class)
+                    .using("email", email)
+                    .load();
```
Comment (Sofasmile) [inline src/main/java/dao/impl/hibernate/UserDAOHibImpl.java]: read about difference between load() and get()

File: src/main/java/model/Basket.java [hunk 4]
```diff
@@ -1,13 +1,41 @@
...
+import javax.persistence.ManyToMany;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
 
+@Entity
+@Table(name = "baskets")
```
Comment (Sofasmile) [inline src/main/java/model/Basket.java]: better create table names without plural

File: src/main/java/dao/impl/hibernate/BasketDAOHibImpl.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            logger.error("Error when try add basket for user in DB from hibernate", e);
+        }
+    }
+
+    @Override
+    public List<Basket> getAllBaskets() {
```
[the remark this answers is not in the document]
Comment (ymilenin) [reply src/main/java/dao/impl/hibernate/BasketDAOHibImpl.java]: I don`t  use this method when using DB, I use this method when using in-memory storage so I made a stub

**Pull request merged**